### ISSUE TITLE
Support OpsGenie Heartbeat

### ISSUE THREAD

**Issue by fernandrone** [prometheus-operator/prometheus-operator#3970]
**What is missing?**

As of now, the Prometheus Operator AlertmanagerConfig does not support [OpsGenie Heartbeat](https://docs.opsgenie.com/docs/heartbeat-api) integration.

AlertManager already supports it (see https://github.com/prometheus/alertmanager/pull/444#issuecomment-428493861), but this configuration does not work on the Operator because the Operator [does not accept](https://github.com/prometheus-operator/prometheus-operator/blob/a739cabb839b005c881695702216fbc75529d5f5/pkg/alertmanager/amcfg.go#L809-L811) a basic auth with empty username (more details below).

**Why do we need it?**

A "Heartbeat" (also known as Watchdog/DeadManSwitch in other platforms) is an important feature to monitor if the Prometheus cluster is online. Futhermore, the integration works when using Prometheus "natively", in fact I'm pretty sure it worked in older versions of the Prometheus Operator, before the introduction of the AlertmanagerConfig CRD. Since OpsGenie is one of the most widely used on-call platforms this seems important.

**Environment**

* Prometheus Operator version: quay.io/prometheus-operator/prometheus-operator:v0.44.1

**Anything else we need to know?**:

OpsGenie Heartbeat API supports [two forms of authentication, as seen in their docs](https://docs.opsgenie.com/docs/authentication).

The first one uses a GenieKey header:

```text
Header Key: Authorization
Header Value: GenieKey $apiKey
```

And the second one is a sort of BasicAuth, but with an empty username:

```text
Header Key: Authorization
Header Value: Basic base64(:$apiKey)
```

This poses some problems when we look at the current AlertmanagerConfig [HTTPConfig](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#alertmanagerconfig).

It offers two methods for configuring authentication, but none of them work for this situation:

1. `bearerTokenSecret` assumes a `Bearer` token, which doesn't really fit (I tested it and it didn't work anyway, and I believe [I correctly tracked the code back to this line](https://github.com/prometheus/common/blob/6e540be855969b2431824fe4efbd7a3ecee69318/config/http_config.go#L265), where it hardcodes the `Bearer `  prefix, as expected)

2. [basicAuth](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#basicauth) _should work_ but Prometheus Operator [introduced a new check that silently skips the Basic Auth in case either username or password is not defined](https://github.com/prometheus-operator/prometheus-operator/blob/a739cabb839b005c881695702216fbc75529d5f5/pkg/alertmanager/amcfg.go#L809-L811). Which I guess makes sense in most cases but OpsGenie auth does expect an empty username, and that's how it used to be configured:

```yaml
receivers:
 - name: deadmansswitch
   webhook_configs:
   - url: 'https://api.opsgenie.com/v2/heartbeats/HEARTBEAT_NAME/ping'
     send_resolved: true
     http_config:
       basic_auth:
         password: OPS-GENIE-API-KEY
```

I'd open a PR for this, but my suggested fix is simply removing [these lines]([https://github.com/prometheus-operator/prometheus-operator/blob/a739cabb839b005c881695702216fbc75529d5f5/pkg/alertmanager/amcfg.go#L809-L811]) to allow for empty usernames/passwords, what do you think?

**Comment by jw-s** [comment]
Hi all! I was playing around with the config to try and get the heartbeats with basic auth to work and I managed it!

For anyone authenticating to opsgenie via basic auth use the following workaround until a more robust solution is available.

tldr; Use a colon ':' as the username value.

```yaml
#alertmanagerconfig.yaml  
- name: deadMansSwitch_{{.name}}
    webhookConfigs:
    - url: 'https://api.opsgenie.com/v2/heartbeats/HEARTBEAT/ping'
      sendResolved: true
      httpConfig:
        basicAuth:
          username:
            key: username
            name: opsgenie
          password:
            key: apiKey
            name: opsgenie
```

```yaml
#secret.yaml
---
apiVersion: v1
data:
  apiKey: WHATEVER
  username: Og== # this is a ':'
kind: Secret
metadata:
  name: opsgenie
type: Opaque
```

 if you specify the username as a ':' colon it works. I guess because basic auth is specified as username:password and they split these two parts.

**Comment by soloradish** [comment]
#4110 already merged into release branch, version `v0.22+` user can use  config like this
```
      - name: heartbeat-receiver
        webhook_configs:
        - url: 'https://api.opsgenie.com/v2/heartbeats/HEART_BEAT_NAME/ping'
          send_resolved: true
          http_config:
            authorization:
              type: GenieKey
              credentials: 'b39****-****-****-****-********98ac'
```

**Comment by maikelvl** [comment]
@soloradish I think the `send_resolved` can be left out because this alert will always be firing
